src/ripple/rpc/handlers/AccountTx.cpp

```diff
@@ -326,10 +325,29 @@ populateJsonResponse(
                 {
                     Json::Value& jvObj = jvTxns.append(Json::objectValue);
 
-                    jvObj[jss::tx] = txn->getJson(JsonOptions::include_date);
+                    auto const json_tx =
+                        (context.apiVersion > 1 ? jss::tx_json : jss::tx);
+                    if (context.apiVersion > 1)
+                    {
+                        std::string hash;
+                        jvObj[json_tx] = txn->getJson(
+                            JsonOptions::include_date,
+                            false,
+                            false,
+                            {std::ref(hash)});
+                        jvObj[jss::hash] = hash;
+                        jvObj[jss::ledger_index] = txn->getLedger();
+                        jvObj[jss::ledger_hash] =
```

Comment:
@Bronek I felt this would be a good place to add the `jss::close_time_iso` field because other fields are also populated here.

But I get the rationale about fencing it behind `txMeta` 👍

Reply:
Done that now, thanks !

Reply:
On a similar note, I feel line 360 (in this file -- setting of the `jss::validated` field) can be pulled out and placed here (along with other fields like `ledger_index`, `ledger_hash`, etc)

Reply:
actually it needs to be moved - thanks for spotting

Reply:
did you move this field?  I can't spot it for this if-block

Reply:
it's above the `if` block now, as not to break API version 1 users of this method.

Reply:
thanks, github hadn't updated the code 😅  👍